lms/djangoapps/badges/backends/badgr.py

```diff
@@ -166,11 +179,106 @@ def _create_assertion(self, badge_class, user, evidence_url):
         return assertion
 
     @staticmethod
-    def _get_headers():
+    def _fernet_setup():
+        """
+        Set up the Fernet class for encrypting/decrypting tokens,
+        use the SECRET_KEY for creating the encryption key.
+        """
+        fernet_key = base64.urlsafe_b64encode(
+            settings.SECRET_KEY.ljust(64).encode('utf-8')[:32]
```

Comment:
This may warrant a short comment as to _why_ the Fernet key is constructed from the `SECRET_KEY` in this specific manner (namely, that Fernet keys _must always be_ URL-safe base64 encoded 32-byte binary strings).